File: backend/src/main/java/ma/zs/alc/service/facade/collaborator/course/CoursCollaboratorService.java
package ma.zs.alc.service.facade.collaborator.course;

import java.util.List;
import ma.zs.alc.bean.core.course.Cours;
import ma.zs.alc.dao.criteria.core.course.CoursCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface CoursCollaboratorService extends  IService<Cours,CoursCriteria>  {

    List<Cours> findByParcoursId(Long id);
    int deleteByParcoursId(Long id);
    long countByParcoursCode(String code);




}

File: frontend/src/app/shared/service/collaborator/course/CoursCollaborator.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {CoursDto} from '../../../model/course/Cours.model';
import {CoursCriteria} from '../../../criteria/course/CoursCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";
import {Observable} from "rxjs";


@Injectable({
    providedIn: 'root'
})
export class CoursCollaboratorService extends AbstractService<CoursDto, CoursCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'collaborator/cours/';
    }

    public constrcutDto(): CoursDto {
        return new CoursDto();
    }

    public constrcutCriteria(): CoursCriteria {
        return new CoursCriteria();
    }

    findByParcoursId(id: number):Observable<CoursDto[]> {
        return this.http.get<CoursDto[]>(this.API+'parcours/id/'+id);
    }
}
